pandas/_libs/lib.pyx

```diff
@@ -2526,3 +2527,21 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def from_nested_dict(object data) -> dict:
+    cdef:
+        object new_data = collections.defaultdict(dict)
```

Comment:
can you set ``new_data`` to type ``dict new_data`` and make related changes below (use ``dict.setdefault`` etc.)? I think that should make fewer calls into Python, making this faster.